com.unity.visualeffectgraph/Editor/ShaderGraph/VFXShaderGraphParticleOutput.cs

```diff
@@ -3,26 +3,144 @@
+    [CanEditMultipleObjects]
+    class VFXShaderGraphParticleOutputEditor : VFXContextEditor
+    {
+        private MaterialEditor m_MaterialEditor = null;
+
+        private bool m_RequireUpdateMaterialEditor = false;
+
+        private void RequireUpdateMaterialEditor() => m_RequireUpdateMaterialEditor = true;
+
+        protected new void OnEnable()
+        {
+            UpdateMaterialEditor();
+            foreach (VFXShaderGraphParticleOutput output in targets)
+            {
+                if (output != null)
+                    output.OnMaterialChange += RequireUpdateMaterialEditor;
+            }
+
+            base.OnEnable();
+        }
+
+        protected new void OnDisable()
+        {
+            foreach (VFXShaderGraphParticleOutput output in targets)
+            {
+                if (output != null)
+                    output.OnMaterialChange -= RequireUpdateMaterialEditor;
+            }
+
+            DestroyImmediate(m_MaterialEditor);
+            base.OnDisable();
+        }
+
+        void UpdateMaterialEditor()
+        {
+            var material = ((VFXShaderGraphParticleOutput)target).transientMaterial;
+            m_MaterialEditor = (MaterialEditor)CreateEditor(material);
+        }
+
+        public override void OnInspectorGUI()
+        {
+            if (targets.OfType<VFXShaderGraphParticleOutput>().Any(context => context.GetOrRefreshShaderGraphObject() == null))
+            {
+                base.OnInspectorGUI();
+                return;
+            }
+
+            serializedObject.Update();
+
+            if (m_RequireUpdateMaterialEditor)
+            {
+                UpdateMaterialEditor();
+                m_RequireUpdateMaterialEditor = false;
+            }
+
+            var materialChanged = false;
+
+            var previousBlendMode = ((VFXShaderGraphParticleOutput)target).GetMaterialBlendMode();
+
+            if (m_MaterialEditor != null)
+            {
+                if (m_MaterialEditor.target == null || (m_MaterialEditor.target as Material)?.shader == null)
+                {
+                    EditorGUILayout.HelpBox("Material Destroyed.", MessageType.Warning);
+                }
+                else
+                {
+                    using (new EditorGUI.DisabledScope(true))
+                    {
+                        // Required to draw the header to draw OnInspectorGUI.
+                        m_MaterialEditor.DrawHeader();
+                    }
+
+                    EditorGUI.BeginChangeCheck();
+
+                    // This will correctly handle the configuration of keyword and pass setup.
+                    m_MaterialEditor.OnInspectorGUI();
+
+                    materialChanged = EditorGUI.EndChangeCheck();
+                }
+            }
+
+            // Indicate caution to the user if transparent motion vectors are disabled and motion vectors are enabled.
+            if ((m_MaterialEditor.target != null) &&
```

Comment:
Thi code should be in the previous `if (m_MaterialEditor != null)` scope

Reply:
Indeed, and @vlad-unity caught an issue caused by it here: https://docs.google.com/document/d/1DkYmO_sQsRaAaB0s4DI9QxXlDqmoK5jt1xs8WyE78Ig/edit?disco=AAAALpPkIcs

It should now be resolved.